### PR TITLE
optimize: add rolling horizon function

### PR DESCRIPTION
**Pull request by FabianHofmann**
Add integrated rolling horizon function, call it with 

```
n.optimize.optimize_with_rolling_horizon()
```

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

File: pypsa/optimization/abstract.py [hunk 1]
```diff
@@ -248,3 +248,30 @@ def optimize_security_constrained(
...
+    Returns
+    -------
+    None
+    """
+    if snapshots is None:
+        snapshots = n.snapshots
+
+    n_snapshots = len(snapshots)
+    for i in range(0, len(snapshots), horizon):
+        n.optimize(snapshots[i : min(n_snapshots, i + horizon)], **kwargs)
```
Comment (fneum) [inline pypsa/optimization/abstract.py:276]: One should pass the state of charges from previous optimisation horizons:

https://github.com/PyPSA/pypsa-eur/pull/204/files#diff-5c213776d6b3578d6249226cef589c81d6f7846a0c7e2f22d05da726400d3875

Comment (FabianHofmann) [reply pypsa/optimization/abstract.py:276]: thanks, will take care of it

**Comment by fneum** [comment]
One possible addition:

In addition to the optimization horizon, this function could also take a window overlap as an argument (i.e. solve week 1+2, then 2+3 based on state after week 1, then 3+4 etc.).

https://github.com/PyPSA/pypsa-eur/blob/e8772c33401d8e5ba4290678d820b18004f0d6d4/scripts/solve_operations_network_other_year_myopic.py#L73-L104

File: doc/release_notes.rst [hunk 2]
```diff
@@ -11,10 +11,8 @@ Upcoming Release
   now allows the passing of a list of buses for which aggregation of __all__ carriers is desired. Generation from
   a carrier at a bus is aggregated now if: It is __either__ in the passed list of aggregated carriers, __or__ in the
   list of aggregated buses.
-* PyPSA now supports stand-by cost terms. A new column
-  `stand_by_cost` was added to generators and links. The stand-by
-  cost is added to the objective function when calling
-  ``n.optimize()``.
+* The ``n.optimize`` accessor now provides a function ``optimize_with_rolling_horizon`` which splits whole optimization of the whole time-span into multiple subproblems which are solved consecutively. This is useful for operational optimizations with a high spatial resolution.
```
Comment (fneum) [inline doc/release_notes.rst:14]:
```suggestion
* The ``n.optimize`` accessor now provides functionality for rolling horizon optimisation using ``n.optimize.optimize_with_rolling_horizon()`` which splits whole optimization of the whole time span into multiple subproblems which are solved consecutively. This is useful for operational optimizations with a high spatial resolution.
```

**Review by fneum — APPROVED** [review]
Tested and LGTM.

**Pull request merged**